apm-agent-plugins/apm-httpclient11-plugin/src/main/java/co/elastic/apm/agent/httpclient/HttpClientInstrumentation.java

```diff
@@ -0,0 +1,107 @@
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.httpclient;
+
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.impl.transaction.AbstractSpan;
+import co.elastic.apm.agent.impl.transaction.Span;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.NamedElement;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.nameContains;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+
+public class HttpClientInstrumentation extends AbstractHttpClientInstrumentation {
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return HttpClient11Advice.class;
+    }
+
+    @VisibleForAdvice
+    public static class HttpClient11Advice {
+
+        @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
+        public static void onBeforeExecute(@Advice.Argument(value = 0) HttpRequest httpRequest) {
+            if (tracer.getActive() == null) {
+                return;
+            }
+            final AbstractSpan<?> parent = tracer.getActive();
+            URI uri = httpRequest.uri();
+            Span span = co.elastic.apm.agent.http.client.HttpClientHelper.startHttpClientSpan(parent, httpRequest.method(), uri.toString(), uri.getScheme(),
+                HttpClientHelper.computeHostName(uri.getHost()), uri.getPort());
+            if (span != null) {
+                span.activate();
+            }
+        }
+
+        @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class, inline = false)
+        public static void onAfterExecute(@Advice.Return @Nullable HttpResponse response,
+                                          @Advice.Thrown @Nullable Throwable t) {
+            final Span span = tracer.getActiveExitSpan();
+            if (span != null) {
```

Comment:
nit: better return here to reduce the level of nesting

Reply:
fixed